src/config.ts

```diff
@@ -116,7 +114,19 @@ export class Configuration {
     return this._initialDevelopment;
   }
 
-  private loadFromData(data: IConfiguration): void {
+  setRuleActivationStatus(ruleId: string, enabled: boolean): void {
```

Comment:
The `rules` object was defined with a different type than its actual usage, leading to inconsistent behavior (one exception in particular) due to mismatching type expectations.

Ensured that we are using the `Map`-type correctly and added a wrapper function to avoid updating settings for undefined rules

Reply:
Will opt for `setRuleActive()` :+1: